From: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Date: Fri, 5 Oct 2012 11:20:10 +0000
Subject: [PATCH] man-dir

===================================================================
---
 Makefile.org |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.org b/Makefile.org
index 761461d..035c03a 100644
--- a/Makefile.org
+++ b/Makefile.org
@@ -153,7 +153,7 @@ TESTS = alltests
 
 MAKEFILE= Makefile
 
-MANDIR=$(OPENSSLDIR)/man
+MANDIR=/usr/share/man
 MAN1=1
 MAN3=3
 MANSUFFIX=
-- 
