From: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Date: Fri, 5 Oct 2012 11:20:10 +0000
Subject: [PATCH] no-symbolic

===================================================================
---
 Makefile.shared |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.shared b/Makefile.shared
index 6951881..48e2710 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -149,7 +149,7 @@ DO_GNU_SO=$(CALC_VERSIONS); \
 	SHLIB_SUFFIX=; \
 	ALLSYMSFLAGS='-Wl,--whole-archive'; \
 	NOALLSYMSFLAGS='-Wl,--no-whole-archive'; \
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
+	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
 
 DO_GNU_APP=LDFLAGS="$(CFLAGS)"
 
-- 
