From: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Date: Fri, 5 Oct 2012 11:20:11 +0000
Subject: [PATCH] shared-lib-ext

===================================================================
---
 Configure |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Configure b/Configure
index 3646248..9b21cdc 100755
--- a/Configure
+++ b/Configure
@@ -1662,7 +1662,8 @@ while (<IN>)
 	elsif ($shared_extension ne "" && $shared_extension =~ /^\.s([ol])\.[^\.]*\.[^\.]*$/)
 		{
 		my $sotmp = $1;
-		s/^SHARED_LIBS_LINK_EXTS=.*/SHARED_LIBS_LINK_EXTS=.s$sotmp.\$(SHLIB_MAJOR) .s$sotmp/;
+#		s/^SHARED_LIBS_LINK_EXTS=.*/SHARED_LIBS_LINK_EXTS=.s$sotmp.\$(SHLIB_MAJOR) .s$sotmp/;
+		s/^SHARED_LIBS_LINK_EXTS=.*/SHARED_LIBS_LINK_EXTS=.s$sotmp/;
 		}
 	elsif ($shared_extension ne "" && $shared_extension =~ /^\.[^\.]*\.[^\.]*\.dylib$/)
 		{
-- 
